stacks-and-queues/lib/fifo_animal_shelter.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+const Queue = require('../lib/queue');
+
+const _validate = Symbol('validate');
+
+class AnimalShelter extends Queue {
+  constructor() {
+    super();
+  }
+
+  enqueue(value) {
+    this[_validate](value);
+
+    super.enqueue(value);
+  }
+
+  [_validate](value) {
+    if (typeof value !== 'string') {
+      throw new Error('Input Error: enqueue requires a string "cat" or "dog"');
+    }
+
+    value.toLowerCase();
+    if (value === 'dog' || value === 'cat') {
+      return;
+
+    } else {
+      throw new Error('Input Error: enqueue requires a string "cat" or "dog"');
+    }
+
+    /* 
+    * Any ideas why this condition would resolve to true and throw my error
+    * with value 'cat'?
+    * 
+    * if (value !== 'dog' || value !== 'cat') {
+    *   throw new Error('Input Error: enqueue requires a string "cat" or "dog"');
+    * }
+    * 
+    */
+  }
+
+  dequeue(pref) {

```

Comment:
If I enter a non-cat or non-dog, such as `dequeue('hello')`, the error that gets thrown is 'Error: No nodes to remove`. In the case of an animal shelter, what are nodes? This error message does not make contextual sense for an animal shelter. It'd be better to have a new error handler that says `No such animal` or something along those lines.